Makefile

```diff
@@ -0,0 +1,484 @@
+
+# Grouped Kotlin bindings
+.PHONY: bindings-kotlin
+bindings-kotlin: bindings-kotlin-android bindings-kotlin-jvm
+
+#-------------------------------------------------------------------------------
+# 4) WASM build + JS deps via stamps
+#-------------------------------------------------------------------------------
+
+# The default build condition is `dev`, which is much faster.
+# To enable optimizations, set the `WASM_RELEASE` variable to any non-empty value:
+#
+# i.e.
+#
+#   make wasm WASM_RELEASE=1
+ifeq ($(strip $(WASM_RELEASE)),)
+  WASM_BUILD_ARGS := --dev
+else
+  WASM_BUILD_ARGS :=
+endif
+
+# Generate WASM package
+wasm-build.stamp: FORCE
+	cd crypto-ffi && \
+	$(WASM_PACK) build \
+	  --locked \
+	  --no-pack \
+	  --out-dir bindings/js/src/autogenerated \
+	  --out-name core-crypto-ffi \
+	  --mode normal \
+	  --target web \
+	  $(WASM_BUILD_ARGS)
+	touch $@
+
+.PHONY: wasm-build
+wasm-build: wasm-build.stamp
+
+# Install JS deps with Bun
+bun-deps.stamp: $(PACKAGE_JSON)
+	$(BUN) install --frozen-lockfile --cwd crypto-ffi/bindings/js
+	touch $@
+
+.PHONY: bun-deps
+bun-deps: bun-deps.stamp
+
+# Full JS binding step
+bindings-js: wasm-build $(JS_OUT)
+
+# All bindings
+.PHONY: bindings
+bindings: bindings-swift bindings-kotlin bindings-js
+
+#-------------------------------------------------------------------------------
+# 5) Documentation targets (stamp-based if desired)
+#-------------------------------------------------------------------------------
+
+# Rust generic docs
+docs-rust-generic.stamp: FORCE
+	$(CARGO) doc --no-deps
+	touch $@
+
+.PHONY: docs-rust-generic
+docs-rust-generic: docs-rust-generic.stamp
+
+# Rust WASM docs
+docs-rust-wasm.stamp: FORCE
+	$(CARGO) doc --no-deps --target=wasm32-unknown-unknown
+	touch $@
+
+.PHONY: docs-rust-wasm
+docs-rust-wasm: docs-rust-wasm.stamp
+
+# Kotlin Dokka (Android) docs
+docs-kotlin.stamp: jvm
+	cd crypto-ffi/bindings && ./gradlew android:dokkaGeneratePublicationHtml
+	mkdir -p target/kotlin/doc
+	cp -R crypto-ffi/bindings/android/build/dokka/html/ target/kotlin/doc
+	touch $@
+
+.PHONY: docs-kotlin
+docs-kotlin: docs-kotlin.stamp
+
+# TypeScript docs via Typedoc
+docs-ts.stamp: wasm-build.stamp bun-deps.stamp
+	cd crypto-ffi/bindings/js && \
+	$(BUN) x typedoc \
+	  --basePath ./ \
+	  --entryPoints src/CoreCrypto.ts \
+	  --tsconfig tsconfig.json \
+	  --out ../../target/typescript/doc \
+	  --readme none \
+	  --treatWarningsAsErrors
+	touch $@
+
+.PHONY: docs-ts
+docs-ts: docs-ts.stamp
+
+# Swift docs via Jazzy (macOS only)
+docs-swift.stamp:
+	mkdir -p target/swift/doc
+	cd crypto-ffi/bindings/Swift/WireCoreCrypto && \
+	jazzy \
+	  --modules WireCoreCrypto,WireCoreCryptoUniffi \
+	  --build-tool-arguments "-project WireCoreCrypto.xcodeproj -scheme WireCoreCrypto" \
+	  -o ../../../target/swift/doc
+	touch $@
+
+.PHONY: docs-swift
+docs-swift: docs-swift.stamp
+
+# Group all docs
+.PHONY: docs
+docs: docs-rust-generic docs-rust-wasm docs-kotlin docs-ts docs-swift
```

Comment:
`docs-swift` should be skipped on non-Apple machines because the bindings and jazzy are not available.